### PR TITLE
TELCODOCS-1146: Installing a cluster on bare metal with AI

### PR DESCRIPTION
**Pull request by johnwilkins**
Revised installation introduction to include Assisted Installer. Revised additional docs to include AI for Nutanix and vSphere.

Fixes: [TELCODOCS-1146](https://issues.redhat.com//browse/TELCODOCS-1146)

See https://issues.redhat.com/browse/TELCODOCS-1146 for additional details.

Preview URL: http://jowilkin.com:8080/TELCODOCS-1146/installing/index.html http://jowilkin.com:8080/TELCODOCS-1146/installing/installing_vsphere/preparing-to-install-on-vsphere.html#choosing-a-method-to-install-ocp-on-vsphere http://jowilkin.com:8080/TELCODOCS-1146/installing/installing_nutanix/installing-nutanix-installer-provisioned.html

For release(s): 4.12+
QE Review: 

- [ ] QE has approved this change. 

Signed-off-by: John Wilkins <jowilkin@redhat.com>

### CODE REVIEW

File: installing/installing_bare_metal/preparing-to-install-on-bare-metal.adoc [hunk 1]
```diff
@@ -24,7 +24,7 @@ include::modules/virt-planning-bare-metal-cluster-for-ocp-virt.adoc[leveloffset=
...
+You can install {product-title} with the link:https://access.redhat.com/documentation/en-us/assisted_installer_for_openshift_container_platform/2022/html-single/assisted_installer_for_openshift_container_platform/index[{ai-full}], with the link:https://console.redhat.com/openshift/install/metal/agent-based[agent-based installer], and on installer-provisioned or user-provisioned infrastructure. The {ai-full} is recommended for connected environments, and the agent-based installer is recommended for air-gapped/restricted networks. The default installation type uses installer-provisioned infrastructure, where the installation program provisions the underlying infrastructure for the cluster. You can also install {product-title} on infrastructure that you provision. If you do not use infrastructure that the installation program provisions, you must manage and maintain the cluster resources yourself.
```
[the remark this answers is not in the document]
Comment (nirfarkas) [reply installing/installing_bare_metal/preparing-to-install-on-bare-metal.adoc:27]: Also related:
https://issues.redhat.com/browse/PD-1446

[the remark this answers is not in the document]
Comment (johnwilkins) [reply installing/installing_bare_metal/preparing-to-install-on-bare-metal.adoc]: @racedo How does this look? http://jowilkin.com:8080/TELCODOCS-1146/installing/installing_bare_metal/preparing-to-install-on-bare-metal.html#choosing-a-method-to-install-ocp-on-bare-metal

File: installing/installing_bare_metal/preparing-to-install-on-bare-metal.adoc [hunk 2]
```diff
@@ -24,7 +24,20 @@ include::modules/virt-planning-bare-metal-cluster-for-ocp-virt.adoc[leveloffset=
...
+The {product-title} installation program offers four methods for deploying a cluster:
+
+* *Interactive*: You can deploy a cluster with the web-based link:https://access.redhat.com/documentation/en-us/assisted_installer_for_openshift_container_platform/2022/html-single/assisted_installer_for_openshift_container_platform/index[{ai-full}]. This is the recommended approach for clusters with networks connected to the internet. The {ai-full} is the easiest way to install {product-title}, it provides smart defaults, and it performs pre-flight validations before installing the cluster. It also provides a RESTful API for automation and advanced configuration scenarios.
+
+* *Local Agent-based*: You can deploy a cluster locally with the agent-based installer for air-gapped/restricted networks. It provides many of the benefits of the {ai-full}, but you must download and configure the link:https://console.redhat.com/openshift/install/metal/agent-based[agent-based installer] first. Configuration is done with a commandline interface. This approach is ideal for air-gapped/restricted networks.
```
Comment (adellape) [inline installing/installing_bare_metal/preparing-to-install-on-bare-metal.adoc:31]: Init cap "Agent" gave me pause but looks like that's how it is in the console for some reason, so 🤷.

Comment (adellape) [inline installing/installing_bare_metal/preparing-to-install-on-bare-metal.adoc:31]: Suggest removing the `/`: "air-gapped or restricted networks"

Apply throughout if updated.

File: modules/installation-process.adoc [hunk 3]
```diff
@@ -20,11 +20,15 @@ endif::[]
...
 
-* For clusters with installer-provisioned infrastructure, you delegate the infrastructure bootstrapping and provisioning to the installation program instead of doing it yourself. The installation program creates all of the networking, machines, and operating systems that are required to support the cluster.
+* To deploy a cluster with the {ai-full}, you configure the cluster settings using the link:https://access.redhat.com/documentation/en-us/assisted_installer_for_openshift_container_platform/2022/html-single/assisted_installer_for_openshift_container_platform/index[{ai-full}]. There is no installer to download and configure. Once you complete the configuration, you download a discovery ISO and boot cluster machines with that image. You can install clusters with the {ai-full} on Nutanix, vSphere, and bare metal with full integration, and other platforms without integration. If you install on bare metal, you must provide all of the cluster infrastructure and resources, including the networking, load balancing, storage, and individual cluster machines.
```
Comment (adellape) [inline modules/installation-process.adoc:23]: s/Once/After/

File: modules/installation-process.adoc [hunk 4]
```diff
@@ -20,11 +20,15 @@ endif::[]
...
+
+* To deploy clusters with the agent-based installer, you download the link:https://console.redhat.com/openshift/install/metal/agent-based[agent-based installer] first. Then you configure the cluster and generate a discovery image. You boot cluster machines with the discovery image, which installs an agent that communicates with the installation program and handles the provisioning for you instead of you interacting with the installation program or setting up a provisioner machine yourself. You must provide all of the cluster infrastructure and resources, including the networking, load balancing, storage, and individual cluster machines. This approach is ideal for air-gapped/restricted network environments.
```
Comment (adellape) [inline modules/installation-process.adoc:25]: s/Then/Then, /

File: modules/installation-process.adoc [hunk 5]
```diff
@@ -40,6 +44,22 @@ The installation configuration files are all pruned when you run the installatio
 You cannot modify the parameters that you set during installation, but you can modify many cluster attributes after installation.
 ====
 
+[discrete]
+== The installation process with the {ai-full}
+
+Installation with the link:https://access.redhat.com/documentation/en-us/assisted_installer_for_openshift_container_platform/2022/html-single/assisted_installer_for_openshift_container_platform/index[{ai-full}] involves creating a cluster configuration interactively using the web-based user interface or using the RESTful API. The {ai-full} user interface prompts you for required values and provides reasonable default values for the remaining parameters, unless you change them in the user interface or with the API.  The {ai-full} generates a discovery image, which you download and use to boot the cluster machines. The image installs RHCOS and an agent, and the agent handles the provisioning for you. You can install {product-title} with the {ai-full} and full integration on Nutanix, vSphere and bare metal, and on other platforms without integration.
```
Comment (adellape) [inline modules/installation-process.adoc:50]: Could "RHCOS" be swapped for one of the related common attributes?

```
:op-system-first: Red Hat Enterprise Linux CoreOS (RHCOS)
:op-system: RHCOS
```

Comment (adellape) [inline modules/installation-process.adoc:50]: Comma after "vSphere"?

**Comment by adellape** [comment]
And reminder to squash down to 1 commit before merge. 🚀

File: installing/installing_bare_metal/preparing-to-install-on-bare-metal.adoc [hunk 6]
```diff
@@ -24,7 +24,20 @@ include::modules/virt-planning-bare-metal-cluster-for-ocp-virt.adoc[leveloffset=
...
+
+* *Local Agent-based*: You can deploy a cluster locally with the agent-based installer for air-gapped or restricted networks. It provides many of the benefits of the {ai-full}, but you must download and configure the link:https://console.redhat.com/openshift/install/metal/agent-based[agent-based installer] first. Configuration is done with a commandline interface. This approach is ideal for air-gapped or restricted networks.
+
+* *Automated*: You can deploy a cluster on installer-provisioned infrastructure and the cluster it maintains. The installer uses each cluster host's baseboard management controller (BMC) for provisioning. You can deploy clusters with both connected or air-gapped or restricted networks.
```
Comment (racedo) [inline installing/installing_bare_metal/preparing-to-install-on-bare-metal.adoc:33]: I would link here the "Deploying installer-provisioned clusters on bare metal" section.

File: installing/installing_bare_metal/preparing-to-install-on-bare-metal.adoc [hunk 7]
```diff
@@ -24,7 +24,20 @@ include::modules/virt-planning-bare-metal-cluster-for-ocp-virt.adoc[leveloffset=
...
+The {product-title} installation program offers four methods for deploying a cluster:
+
+* *Interactive*: You can deploy a cluster with the web-based link:https://access.redhat.com/documentation/en-us/assisted_installer_for_openshift_container_platform/2022/html-single/assisted_installer_for_openshift_container_platform/index[{ai-full}]. This is the recommended approach for clusters with networks connected to the internet. The {ai-full} is the easiest way to install {product-title}, it provides smart defaults, and it performs pre-flight validations before installing the cluster. It also provides a RESTful API for automation and advanced configuration scenarios.
+
+* *Local Agent-based*: You can deploy a cluster locally with the agent-based installer for air-gapped or restricted networks. It provides many of the benefits of the {ai-full}, but you must download and configure the link:https://console.redhat.com/openshift/install/metal/agent-based[agent-based installer] first. Configuration is done with a commandline interface. This approach is ideal for air-gapped or restricted networks.
+
```
Comment (racedo) [inline installing/installing_bare_metal/preparing-to-install-on-bare-metal.adoc:32]: I would link here the section for agent-based installer "Installing an on-premise cluster with the Agent-based installer" also.

File: installing/installing_bare_metal/preparing-to-install-on-bare-metal.adoc [hunk 8]
```diff
@@ -24,7 +24,20 @@ include::modules/virt-planning-bare-metal-cluster-for-ocp-virt.adoc[leveloffset=
...
+* *Automated*: You can deploy a cluster on installer-provisioned infrastructure and the cluster it maintains. The installer uses each cluster host's baseboard management controller (BMC) for provisioning. You can deploy clusters with both connected or air-gapped or restricted networks.
+
+* *Full control*: You can deploy a cluster on infrastructure that you prepare and maintain, which provides maximum customizability. You can deploy clusters with both connected or air-gapped or restricted networks.
+
```
Comment (racedo) [inline installing/installing_bare_metal/preparing-to-install-on-bare-metal.adoc:36]: And link here the UPI section "Installing a user-provisioned cluster on bare metal"

Links to each of these 4 sections would tie it together better for users IMO

**Comment by racedo** [comment]
Would it make sense to place the "Deploying installer-provisioned clusters on bare metal" sections under "Installing on bare metal". As a user of these docs I find it a bit confusing to have a section to install bare metal with all the options except one which has its own separate section.

**Comment by johnwilkins** [comment]
@adellape Can you indicate peer review done?

**Pull request merged**